File: gothello-api/gothello/stone.py
from enum import Enum

class Stone(Enum):
    NONE = 0
    BLACK = 1
    WHITE = 2
    DRAW = 3
    SPECTATOR = 4
    LEGAL = 5
    ILLEGAL = 6

    @classmethod
    def from_char(cls, char):
        if char == "I":
            return Stone.ILLEGAL
        if char == "L":
            return Stone.LEGAL
        if char == "B":
            return Stone.BLACK
        if char == "W":
            return Stone.WHITE

File: gothello-server/src/main/java/gothello/gothelloserver/rules/Stone.java
package gothello.gothelloserver.rules;

/**
 * Stone
 */
public enum Stone {

    BLACK("black", true),
    WHITE("white", true),
    DRAW("draw", false),
    SPECTATOR("spectator", false),
    NONE("none", false);

    private final String name;
    private final boolean playable;

    private Stone(String name, boolean playable) {
        this.name = name;
        this.playable = playable;
    }

    public String toString() {
        return name;
    }

    public Boolean isPlayable() {
        return playable;
    }

    public static Stone otherPlayer(Stone player) {
        switch (player) {
            case WHITE:
                return Stone.BLACK;
            case BLACK:
                return Stone.WHITE;
            default:
                return Stone.WHITE;
        }
    }

}